src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java

```diff
@@ -146,6 +153,13 @@ public String getVersion() {
      */
     public abstract LinkedHashSet<String> getSupportedFuncotationFields();
 
+    /**
+     * @return An ordered {@link LinkedHashSet} of the names of annotations that this Data Source supports when annotating segments.
+     */
+    public LinkedHashSet<String> getSupportedFuncotationFieldsForSegments() {
```

Comment:
I agree it would be cleaner, but we would have to make sure that it supports mixed inputs.  

Filed an issue to investigate a refactoring:  #5961 

No action.